### DISCUSSION TITLE
[PR #3451] Admin Grids - Reset Columns to default arrangement

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#3511] [Ideas]
The open **PR #3451** adds a new useful feature. Based on the product attributes we are able to set custom grids in the Backend by choosing and arranging the columns as we want.

This thread is meant to solve an issue. A custom grid must to be brought to its default arrangement. A button solves this problem to rearrange the columns to the default. Without this feature, we risk that someone will want to go back to the default arrangement and he doesn't know how to do it. Releasing it without a reset button (e.g. [Reset Grid]) would not be right. To be more explicit, the columns can be controlled from the configuration page, but if the default arrangement is changed with the default columns, the grid must be brought back to its initial.

The developer @empiricompany wants a feedback from us.

**Question: Where is the right place to add the reset grid button?** 

There are several places. in the grid page, in the configuration page or in both. All three variants are good options and must be analyzed from the perspective of the user who expects that when he wants to restore the grid to the initial, this task should be intuitively.

Until I do my homework to express a personal opinion with arguments, please feel free to let us know where is the right place for the button.

PS - I chose this approach in the Discussions section because some PRs already have a lot of comments and are hard to follow. I think the best approach from now on is to use this section to highlight changes that are then implemented by the PR authors.

**Comment by empiricompany** [comment]
thanks @ADDISON74
more important than the reset button is its interface implementation, naming conventions, path of new files and other questions related that you can find in the description of the PR

**Reply by addison74** [reply]
Don't expect to get a feedback on that topic too soon. What you mentioned there is the back part that belongs to the developer. the purpose of the discussion is the seen side of things, so that everyone is satisfied.
